dll/win32/shell32/CDefView.cpp

```diff
@@ -587,28 +591,54 @@ void CDefView::UpdateStatusbar()
             }
         }
 
+        szPartText[0] = szPartText2[0] = UNICODE_NULL;
+
         /* Don't show the file size text if there is 0 bytes in the folder
          * OR we only have folders selected. */
         if ((cSelectedItems && !bIsOnlyFoldersSelected) || uTotalFileSize)
-        {
             StrFormatByteSizeW(uTotalFileSize, szPartText, _countof(szPartText));
-        }
-        else
-        {
-            *szPartText = 0;
-        }
 
-        m_pShellBrowser->SendControlMsg(FCW_STATUS, SB_SETTEXT, 1, (LPARAM)szPartText, &lResult);
+        if (hThread != m_hUpdateStatusbarThread)
+            return;
 
         /* If we are in a Recycle Bin folder then show no text for the location part. */
         if (!_ILIsBitBucket(m_pidlParent))
         {
-            LoadStringW(shell32_hInstance, IDS_MYCOMPUTER, szPartText, _countof(szPartText));
+            LoadStringW(shell32_hInstance, IDS_MYCOMPUTER, szPartText2, _countof(szPartText2));
```

Comment:
I don't think I understand why we need yet another variable there. (I know this is related to the comment below, but still...)

Why not having:
```c++
        if (hThread != m_hUpdateStatusbarThread)
            return;

        // Now do all the "UI" update:
        //
        m_pShellBrowser->SendControlMsg(FCW_STATUS, SB_SETTEXT, 1, (LPARAM)szPartText, &lResult);
        //
        // and the bitbucket stuff:
        // if (!_ILIsBitBucket(m_pidlParent)) { /* load string etc. */ }
        m_pShellBrowser->SendControlMsg(FCW_STATUS, SB_SETICON, 2, pIcon, &lResult);
        m_pShellBrowser->SendControlMsg(FCW_STATUS, SB_SETTEXT, 2, (LPARAM)szPartText2, &lResult);
```

Reply:
Fixed.